deps/Makefile

```diff
@@ -109,10 +109,19 @@ JEMALLOC_LDFLAGS=$(LDFLAGS)
 ifneq ($(DEB_HOST_GNU_TYPE),)
 JEMALLOC_CONFIGURE_OPTS += --host=$(DEB_HOST_GNU_TYPE)
 endif
+JEMALLOC_CONFIGURE_OPTS += --with-malloc-conf=narenas:1
+
+JEMALLOC_VERSION = 5.3.0
+JEMALLOC_URL = https://github.com/jemalloc/jemalloc/releases/download/$(JEMALLOC_VERSION)/jemalloc-$(JEMALLOC_VERSION).tar.bz2
 
 jemalloc: .make-prerequisites
 	@printf '%b %b\n' $(MAKECOLOR)MAKE$(ENDCOLOR) $(BINCOLOR)$@$(ENDCOLOR)
+	if [ ! -d "jemalloc" ]; then \
```

Comment:
I like the fact that we remove the jemalloc code inside our repo. BUT maybe we should consider using git submodule instead?

Reply:
makes sense i'm not super critical about how to do that, so whatever makes sense

Reply:
Can we just require that it's installed in the system? That's what we do for OpenSSL.

If it's not installed, we can display an error message saying that they can either install it or use another allocator.

Reply:
Today we have specific configurations for jemalloc that needs to be set at compile time (see `./configure` call below). 
My point of view is that, the allocator is super important for any in memory service, valkey should actively enforce usage of desired allocator version and configs. In addition, having no control over the configs, even if we enforce the jemalloc version, would make it much harder to investigate issues or validate system behaviour

Reply:
In that case, I think we should still keep a vendored copy of Jemalloc in the repo, or otherwise let the user specify it in some way. Downloading deps in build time is a controversial thing that is not easily accepted in critical systems. (I know we have systems that use offline build servers.)

Reply:
makes sense, how about @ranshid proposal to use git submodule?

Reply:
I think submodules is a new complexity too. The source releases in tar.gz packages are not standalone anymore. I assume for debian/fedora source code packages it's not ideal either. For those, you want dependencies in the form of other deb/rpm packages.

Reply:
I would prefer we vendor it, but since it's more standard it will be much easier to keep up to date. We can support the option to use the non-vendored option as well though, which will make some of the distributions happy.

Reply:
@madolson if we de-vendor it, we can't control things like the lq-quantum. Is that OK?

Some alternatives: We can give build instructions about how to best build jemalloc for Valkey, or we can even build it from our Makefile if the user just provides the path to the jemalloc's source code.

Reply:
I hear from the discussion that we should not couple the changes. My plan is to update the PR to include standard jemalloc with the defrag change. In a followup discussion/PR we can discuss if/how we want to change the way we build jemalloc lib, does this makes sense? @ranshid @zuiderkwast @madolson